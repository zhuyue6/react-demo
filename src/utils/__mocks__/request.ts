function get() {
  return Promise.resolve(1)
}
export {
  get 
}