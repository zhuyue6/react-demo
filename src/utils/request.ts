import axios from 'axios'

const http = axios.create({
  baseURL: 'api/',
  timeout: 2000,
  responseType: 'json'
})

export function get(url: string, params: { [index: string]: any }) {
  return http.get(url, {
    params
  })
}

export function post(url: string, params: { [index: string]: any }) {
  return http.post(url, params)
}

export default http