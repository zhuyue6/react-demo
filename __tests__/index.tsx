import {create} from 'react-test-renderer';
import axios from 'axios'
import User from '@/components/user';
import {render} from '@testing-library/react';
import {sum} from '@/utils'
import {get, post} from '@/utils/request'
import React from 'react';

it('adds 1 + 2 to equal 3', () => {
  expect.assertions(1);
  expect(3).toBe(3);
});

it('快照测试 react 组件', ()=> {
  const component = create(<User name="user" class='a' />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('节点测试', ()=> {
  const { queryByText } = render(<User name="user" class='a' />);
  expect(queryByText(/user/i)).toBeTruthy();
})

it('mock 函数', ()=> {
  const mockSum = jest.fn(sum)
  expect(mockSum(1, 2)).toBe(3);
  expect(mockSum.call.length).toBe(1)
})

jest.mock('@/utils/request')
it('mock 模块', ()=> {
  return get('', {}).then((data)=>{
    expect(data).toEqual(1)
  })
})