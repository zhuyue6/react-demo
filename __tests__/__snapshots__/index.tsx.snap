// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 react 组件 1`] = `
<div
  className="a"
>
  user
</div>
`;
